lib/application.ts

```diff
@@ -1388,22 +1405,39 @@ export class SNApplication {
     }
   }
 
-  public async removePasscode(): Promise<void> {
+  /**
+   * @returns whether the passcode was successfuly removed or not
+   */
+  public async removePasscode(): Promise<boolean> {
+    const passcode = await this.challengeService.promptForPasscode(
```

Comment:
I feel like `challengeService.promptForPasscode` should be renamed to `challengeService.promptForCorrectPasscode`, if that's the way it works. Not necessary for this PR but something to keep in mind as reading this code is confusing because we don't even validate the resulting `passcode`.